README.md

```diff
@@ -1,7 +1,103 @@
 Tackle2 Addon Jkube
 ===================
-The addon is based on the [Tackle2](https://github.com/konveyor?q=tackle2&type=all&language=&sort=) project.
+The addon is part of the [Tackle2](https://github.com/konveyor?q=tackle2&type=all&language=&sort=) projects.
 
-This addon generate Kubernetes resources from a Java application.
-It uses the [JKube](https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin) Maven plugin to generate the Kubernetes resources.
+This is an implementation of [Jkube's](https://www.eclipse.org/jkube) `kubernetes-maven-plugin` as a konveyor addon. This addon is a small step towards Application Modernization. It brings your <strong>Java</strong> applications on to Kubernetes by leveraging the tasks required to make your application cloud-native. It provides a tight integration into Maven and benefits from the build configuration already provided. 
 
+This addon focus on two tasks: 
+- Generates container images with flexible and powerful configuration.
+- Generates vanilla Kubernetes descriptors (YAML files).
+
+It leverage the [Kubernetes Maven plugin](https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin)  to generate conatiner image and Kubernetes resources.
+
+## Development
+To browse code - [Open in VSCode](https://open.vscode.dev/konveyor/tackle2-addon-jkube) 
+
+We use [tackle2-hub addon](https://github.com/konveyor/tackle2-hub/tree/main/addon) package to integrate addon with [tackle2-hub](https://github.com/konveyor/tackle2-hub). Here is a good starter template to build an addon - [Test Addon](https://github.com/konveyor/tackle2-hub/tree/main/hack/cmd/addon).
+Tackle Hub handles all requests to an addon via [Task APIs](https://github.com/konveyor/tackle2-hub/blob/main/api/task.go) and forward these requests to desired addon.
+
+
+A simple request for an addon would look like this:
+```bash
+host="${HOST:-localhost:8080/hub}"
+
+# Make a request to hub
+request_cmd="$(curl -i -o - -X POST ${host}/tasks -d \
+'{
+    "name":"Jkube",
+    "state": "Ready",
+    "locator": "jkube",
+    "addon": "jkube",
+    "application": {"id": 2},
+    "data": {}
+}')"
+```
+
+In above request we mention `addon: jkube` as an addon identifier and `application: {id: 2}` states the application against which this addon will be executed. The `data` field here is empty but we can pass data to addon during api request and we use following struct to receive data for the addon.
```

Comment:
Should find a way to include actually creating this application, either via the UI or `curl`.

Reply:
The `data` field here is empty because the JKube addon accepts none? Please be more specific here.

Reply:
I've shown in the demo, how we can create an application
Should I mention all the steps needed to create an application here?

Reply:
Yeah I might have missed that, in Jkube we aren't accepting any data and that's why it's empty

Reply:
I believe you should also explain in text here how to create an application, yes.